client/ayon_core/hosts/maya/plugins/publish/validate_instance_in_context.py

```diff
@@ -64,14 +67,22 @@ def get_invalid(cls, instance):
 
     @classmethod
     def repair(cls, instance):
-        context_folder_path = cls.get_context_folder_path(instance)
+        context_folder_path, context_task = cls.get_context(
+            instance)
         instance_node = instance.data["instance_node"]
         cmds.setAttr(
             "{}.folderPath".format(instance_node),
             context_folder_path,
             type="string"
         )
+        cmds.setAttr(
+            "{}.task".format(instance_node),
+            context_task,
+            type="string"
+        )
```

Comment:
This should actually go through the CreateContext to update the instances to ensure product names update along with the correct product naming template logic. Which is something that this PR https://github.com/ynput/ayon-core/pull/358 solves for the "update instances to context".

Reply:
do you mean something similar to the pesudo code below?
```
host = registered_host()
create_context = CreateContext(host)

folder_entity = get_current_task_entity(fields={"attrib"})

instance_values = {
    "folderPath": folder_entity["attrib"]["folderPath"],
    "task": folder_entity["attrib"]["task"]
}

for instance in create_context.instances:
    for key, value in instance_values.items():
        instance[key] = value

create_context.save_changes()
```